src/error.rs

```diff
@@ -70,7 +86,17 @@ impl Error {
     self.hint.to_owned()
   }
 
-  /// Convenience function for saying a thing failed validation.
+  /// Convenience function for saying a thing failed validation using
+  /// `ErrorCode::BadRequest`.
+  ///
+  /// # Example
+  /// ```rust
```

Comment:
I guess, but I prefer being explicit in the case where we need to repurpose the docs. We can copy/paste to a new markdown file more easily. Also for the case where we might not have rust code in a preformatted block.